lib/utils.js

```diff
@@ -392,27 +412,46 @@ function mergeSettings(target, src) {
       }
     });
   } else {
+    let baseModelAclRank;
     if (target != null && typeof target === 'object') {
       // Add properties from target to dst
       Object.keys(target).forEach(function(key) {
+        if (key === 'acls' && Array.isArray(target[key])) {
+          let targetACLs = target[key];
+          targetACLs.forEach(function(acl) {
+            acl.rank = acl.rank || 1;
+            // capture the base model ACL rank
+            baseModelAclRank = baseModelAclRank || acl.rank;
```

Comment:
Don't we have to find out the maximum rank from target ACLs? IIUC, you code will get the last positive rank from all entries.